src/app/modules/upload-status/users-csv/users-csv.component.ts

```diff
@@ -0,0 +1,98 @@
+import { Component, OnInit, ViewChild, ElementRef } from '@angular/core';
+import { MatPaginator, MatTableDataSource, PageEvent, MatDialog } from '@angular/material';
+import { BulkuploadService } from '../../admin-core';
+import { SelectionModel } from '@angular/cdk/collections';
+import { fromEvent, of, Observable } from 'rxjs';
+import { distinctUntilChanged, map, filter } from 'rxjs/operators';
+
+@Component({
+  selector: 'app-users-csv',
+  templateUrl: './users-csv.component.html',
+  styleUrls: ['./users-csv.component.scss']
+})
+export class UsersCsvComponent implements OnInit {
+  dataSource: MatTableDataSource<any>;
+  displayedColumns: any = [];
+  selection = new SelectionModel(true, []);
+  listing: boolean = false;
+  recordCount: any;
+  columns: any;
+  queryParams = {
+    page: 1,
+    size: 10,
+  };
+  organisationListData: any;
+  @ViewChild('searchInput') searchInput: ElementRef;
+  @ViewChild(MatPaginator) paginator: MatPaginator;
+  constructor(private bulkuploadService: BulkuploadService) { }
+
+  ngOnInit() {
+
+    fromEvent(this.searchInput.nativeElement, 'keyup').pipe(
```

Comment:
Create a directive for this and reuse it everywhere, hence we can prevent the repeating of same code